### PR TITLE
promql: Add irate() function

### PR DESCRIPTION
**Pull request by brian-brazil**
irate is a rate function that only looks at the most
recent two data points, and calucaltes a per-second value
from that. This produces much more granular graphs for
fast moving data, and works sanely across many scrape intervals.

It doesn't do so well for slowly moving data.

### CODE REVIEW

**Comment by fabxc** [comment]
:+1:

File: promql/functions.go [hunk 1]
```diff
@@ -127,6 +127,49 @@ func funcIncrease(ev *evaluator, args Expressions) model.Value {
...
+	resultVector := vector{}
+	for _, samples := range ev.evalMatrix(args[0]) {
+		// No sense in trying to compute a rate without at least two points. Drop
+		// this vector element.
+		if len(samples.Values) < 2 {
+			continue
+		}
+
+		var currentSample, previousSample model.SamplePair
+		for _, sample := range samples.Values {
```
Comment (juliusv) [inline promql/functions.go]: Why iterate through all samples if you can just take `samples.Values[n-2]` and `samples.Value[n-1]` where `n := len(samples.Values)`?

Comment (brian-brazil) [inline promql/functions.go]: Hmm, I'd tried that first time around and couldn't get it working. Works now though.

**Comment by juliusv** [comment]
:+1: and :100: for the function name.

**Pull request merged**